src/index/TextIndexConfig.h

```diff
@@ -0,0 +1,111 @@
+ *
+ * Fields:
+ *  - std::optional<const std::string> wordsFile_: Specifies the path to the
+ *                                                wordsFile.
+ *  - std::optional<const std::string> docsFile_: Specifies the path to the
+ *                                               docsFile.
+ *  - bool addWordsFromLiterals: If set to true the literals from the KB are
+ *                               used to build the text index. During retrieval
+ *                               this leads to literals being returned if they
+ *                               contain the search word.
+ * - bool useDocsFileForVocabulary: If set to true the text index is build using
+ *                                  the words from the documents given in the
+ *                                  docsFile instead of the words from the
+ *                                  wordsFile.
+ * - bool addOnlyEntitiesFromWordsFile: If set to true the wordsFile is only
+ *                                      parsed for entities and not words. This
+ *                                      can be used in combination with
+ *                                      useDocsFileForVocabulary to still have
+ *                                      the functionality to check if a text
+ *                                      contains a certain entity even if only
+ *                                      the docsFile words were used to build.
+ * - TextScoringMetric textScoringMetric: There are 3 scoring options for the
+ *                                        text index. First EXPLICIT. Explicit
+ *                                        scores are given in the wordsFile. If
+ *                                        explicit is used with
+ *                                        useDocsFileForVocabulary all scores
+ *                                        are 0. Second TFIDF. If Tf-Idf is used
+ *                                        an inverted index is build without
+ *                                        needing any parameters. Third BM25. If
+ *                                        BM25 is used the b and k parameters
+ *                                        are used. For details see
+ *                                        TextScoring.h.
+ * - std::pair<float, float> bAndKForBM25: The b and k parameter used for the
+ *                                         BM25 scores. Two things have to hold:
+ *                                         0 <= b <= 1 and 0 <= k.
+ */
+struct TextIndexConfig {
+  std::optional<const std::string> wordsFile_ = std::nullopt;
+  std::optional<const std::string> docsFile_ = std::nullopt;
+  bool addWordsFromLiterals_ = true;
+  bool useDocsFileForVocabulary_ = false;
+  bool addOnlyEntitiesFromWordsFile_ = false;
+  TextScoringMetric textScoringMetric_ = TextScoringMetric::EXPLICIT;
+  std::pair<float, float> bAndKForBM25_ = {0.75f, 1.75f};
+
+  void checkValid() const {
```

Comment:
You can give this struct an explicit constructor that automatically calls `checkValid()` at the end + 
if it is not needed or valid to change the settings after the fact, then the members should be private + should have explicit getter functions as `const &`.

(It is not a struct, the members are coupled).